controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -200,11 +200,9 @@ function fetch($request, $actionArgs = array()) {
 		if ($query->getAssocType() == ASSOC_TYPE_SUBMISSION) {
 
 			$queryDao = DAORegistry::getDAO('QueryDAO');
-			$selectedParticipants = $query->getId() ? $queryDao->getParticipantIds($query->getId()) : array();
-
 			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
-			$assignments = $stageAssignmentDao->getBySubmissionAndStageId($query->getAssocId(), $query->getStageId())
-				->toArray();
+
+			$selectedParticipants = $query->getId() ? $queryDao->getParticipantIds($query->getId()) : array();
```

Comment:
I prefer to place the line where a DAO is used next to the line where it is retrieved from the `DAORegistry`. I think, if we ever remove this line, it will be more likely to be "cleaned up" if it's next to it.

Likewise, you can move `$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');` into the callback function since it's a very lightweight call.

Reply:
Doing these changes to the open reviews pr where I already have these same additions.
There $stageAssignmentDao is used in several places so I am now calling it only once. Of course I can add it before all the calls if that is better.

Reply:
No that's fine. Whatever's best given the wider context.